hudi-common/src/main/java/org/apache/hudi/common/model/AWSDmsAvroPayload.java

```diff
@@ -68,6 +68,20 @@ private Option<IndexedRecord> handleDeleteOperation(IndexedRecord insertValue) t
     return delete ? Option.empty() : Option.of(insertValue);
   }
 
+  @Override
+  public OverwriteWithLatestAvroPayload preCombine(OverwriteWithLatestAvroPayload oldValue) {
```

Comment:
Should we have a new base class for these sub-classes? For e.g. `BaseEventTimeOrderingAvroPayload`

Reply:
I mean we need a base class for the shared `#preCombine` logic.

Reply:
We should revisit the payload class hierarchy so that such payload class may extend 
DefaultHoodieRecordPayload.  This requires work to make sure there is no behavior change.  HUDI-8575 to track.